frontend/src/components/module-viewer/content/modals/EditVideoModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { Box, Flex, Spinner, Text, VStack } from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import ReactPlayer from "react-player";
+import { VideoBlockState } from "../../../../types/ContentBlockTypes";
+import { Modal } from "../../../common/Modal";
+import EditorContext from "../../../../contexts/ModuleEditorContext";
+
+import { TextInput } from "../../../common/TextInput";
+import { EditContentModalProps } from "../../../../types/ModuleEditorTypes";
+
+const InvalidLinkPreview = (): React.ReactElement => {
+  return (
+    <Flex
+      height="350px"
+      justifyContent="center"
+      alignItems="center"
+      background="#383838"
+    >
+      <Text variant="sm" color="white">
+        Invalid Link
+      </Text>
+    </Flex>
+  );
+};
+
+const EditVideoModal = ({
+  block,
+  index,
+  isOpen,
+  onClose,
+}: EditContentModalProps<VideoBlockState>): React.ReactElement => {
+  const [link, setLink] = useState(block.content.link ?? "");
+  const [invalid, setInvalid] = useState(true);
+  const context = useContext(EditorContext);
+
+  if (!context) return <></>;
+  const { dispatch } = context;
+
+  const onSave = () => {
+    if (invalid) {
+      return;
+    }
+    const newBlock = {
+      ...block,
+      content: {
+        ...block.content,
+        link,
+      },
+    };
+
+    dispatch({
+      type: "update-block",
+      value: {
+        index,
+        block: newBlock,
+      },
+    });
+
+    onClose();
+  };
+
+  return (
+    <Modal
+      size="xl"
+      header="Edit Video Component"
+      onConfirm={onSave}
+      onCancel={onClose}
+      isOpen={isOpen}
+      canSubmit={!invalid}
+    >
+      <Flex>
+        <VStack flex="1" alignItems="left">
+          <TextInput
+            label="Link:"
+            defaultValue={link}
+            onChange={(e) => {
+              setInvalid(true);
+              setLink(e);
+            }}
+            isInvalid={invalid}
```

Comment:
It's kind of weird that the validation logic says the field is "invalid" when we haven't done anything (i.e., the link field is empty). Could you change this to match the design on initial load (i.e., "Paste a link ...")